src/main/kotlin/ar/edu/unq/criptop2p/service/TransactionService.kt

```diff
@@ -0,0 +1,123 @@
+                                          transactionSummary)
+    }
+
+    private fun transactionSummary(user: Optional<User>, allUserRequest: List<Request>): List<CurrencyTransactionReportDTO> {
+        val result = arrayListOf<CurrencyTransactionReportDTO>()
+
+        for (symbol in cryptoService.getSymbolList()){
+            val currencyList = allUserRequest.filter{it.getCryptoCurrency().getSymbol() == symbol}
+
+            if (currencyList.isEmpty()) continue
+
+            val totalAmountOperated = totalAmountOperated(user.get(), currencyList)
+            val currentUSDPrice = cryptoService.getPrice(symbol)
+            val currentARGPrice = requestService.getDollarPrice()
+            result.add(CurrencyTransactionReportDTO( symbol,
+                                                    totalAmountOperated,
+                                                    currentUSDPrice,
+                                                    currentUSDPrice * currentARGPrice)
+            )
+        }
+        return result
+    }
+
+    private fun totalOperated(user: User, requests: List<Request>): Pair<Double, Double> {
+        var total = Pair(0.0, 0.0)
+        for (r in requests) {
+            val operated = operated(user, r)
+            total = Pair(total.first + operated.first, total.second + operated.second)
+        }
+        return total
+    }
+
+    fun operated(user: User, request: Request): Pair<Double, Double> {
+        var result = Pair(0.0, 0.0)
+        if (request.getType() == RequestType.BUY) {
+            if (request.getOwner().getId() == user.getId()) {
+                val usdOperated = request.getAmount() * request.getCryptoCurrency().getPrice()
+                result = Pair(usdOperated, usdOperated * request.getPriceArgAtCompletation())
+            }
+            else {
+                var usdOperated = request.getAmount() * request.getCryptoCurrency().getPrice() * -1
+                result = Pair(usdOperated, usdOperated * request.getPriceArgAtCompletation())
+            }
+        } else {
+            if (request.getType() == RequestType.SELL) {
+                if (request.getOwner().getId() == user.getId()) {
+                    var usdOperated = request.getAmount() * request.getCryptoCurrency().getPrice() * -1
+                    result = Pair(usdOperated, usdOperated * request.getPriceArgAtCompletation())
+
+                } else {
+                    var usdOperated = request.getAmount() * request.getCryptoCurrency().getPrice()
+                    result = Pair(usdOperated, usdOperated * request.getPriceArgAtCompletation())
+                }
+            }
+        }
+       return result
+    }
+
+    private fun totalAmountOperated(user: User, requests: List<Request>): Double {
+        var total = 0.0
+        for (r in requests) {
+            total += amountOperated(user, r)
+        }
+        return total
+    }
+
+    fun amountOperated(user: User, request: Request): Double {
```

Comment:
Está lógica la podrías simplificar así:

![image](https://user-images.githubusercontent.com/54821556/196043326-c868b141-e78b-4659-9381-ccf10485f5de.png)
Y esto dentro de la clase Request:

`    fun amountOperated(user: User): Double = this.getType().amountOperated(this, user)`

De está forma también pasaría la lógica al modelo y la podríamos testear más fácil.

Reply:
Listo.